src/js/actions/guides.js

```diff
@@ -313,6 +313,36 @@ define(function (require, exports) {
     setGuide.transfers = [resetGuidePolicies, deleteGuide];
 
     /**
+     * Clears all the guides in the given document
+     *
+     * @param {{id: number}} document Document model or an object containing document ID
+     * @return {Promise}
+     */
+    var clearGuides = function (document) {
+        if (document === undefined) {
+            var appStore = this.flux.store("application");
+
+            document = appStore.getCurrentDocument();
+        }
+
+        var payload = {
+                documentID: document.id
+            },
+            clearObj = documentLib.clearGuides(documentLib.referenceBy.id(document.id)),
+            dispatchPromise = this.dispatchAsync(events.document.history.nonOptimistic.GUIDES_CLEARED, payload),
+            clearPromise = descriptor.playObject(clearObj);
+
+        return Promise.join(dispatchPromise, clearPromise)
+            .bind(this)
+            .then(function () {
+                return this.transfer(resetGuidePolicies);
+            });
+    };
+    clearGuides.reads = [];
+    clearGuides.writes = [locks.JS_DOC];
```

Comment:
this should lock PS_DOC too?  (strictly speaking, even though the transfer probably handles glosses over this fact)